app/code/ZP/LoyaltyProgram/Observer/Customer/Model/ResourceModel/AbstractAssignLoyaltyProgram.php

```diff
@@ -0,0 +1,92 @@
+<?php
+declare(strict_types=1);
+
+namespace ZP\LoyaltyProgram\Observer\Customer\Model\ResourceModel;
+
+use Magento\Customer\Model\Data\Customer;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\StoreManagerInterface;
+use ZP\LoyaltyProgram\Api\LoyaltyProgramManagementInterface;
+use ZP\LoyaltyProgram\Model\Configs\Program\Scope\Config as ProgramScopeConfig;
+use Magento\Customer\Api\Data\CustomerExtensionInterfaceFactory;
+use ZP\LoyaltyProgram\Api\LoyaltyProgramRepositoryInterface;
+use ZP\LoyaltyProgram\Model\Validators\Data\Validator;
+
+abstract class AbstractAssignLoyaltyProgram
+{
+    public function __construct(
+        protected LoyaltyProgramManagementInterface $loyaltyProgramManagement,
+        protected StoreManagerInterface $storeManager,
+        protected ProgramScopeConfig $programScopeConfig,
+        protected CustomerExtensionInterfaceFactory $customerExtensionFactory,
+        protected LoyaltyProgramRepositoryInterface $loyaltyProgramRepository,
+        protected Validator $dataValidator
+    ) {}
+
+    /**
+     * @param Observer $observer
+     * @throws \Exception
+     */
+    public function execute(Observer $observer)
+    {
+        if ($this->isProgramEnabled()) {
```

Comment:
why just bot to use `$this->programScopeConfig->isEnabled((int)$this->storeManager->getWebsite()->getId());`
here? do we really need a separate method for this?